### ISSUE TITLE
webpack.externals relative path causes Unexpected token .

### ISSUE THREAD

**Issue by weijiangan** [webpack/webpack#7357]
# Bug report

**What is the current behavior?**
If relative path is added to `webpack.externals` in the config, it generates the following code that is a syntax error
```js
module.exports = ../my_module;//# sourceURL=[module]
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Add a module that has relative path to `webpack.externals` config, e.g. for my case

```js
{
  externals: '../webpack.config'
...
}
```

**What is the expected behavior?**
It should not bundle the module and require the one in my environment

**Other relevant information:**
webpack version:  4.8.3
Node.js version: 10.0.0
Operating System: macOS 10.13.5

**Comment by ooflorent** [comment]
When `externals` is a string, it produces the following output:

```js
// in this example `externals=jquery`
module.exports = jquery;
```

I admit, the documentation is not crystal clear about how it works. Given the following external definition:

```js
externals: {
  "../webpack.config": "cfg",
}
```

webpack will assume that `../webpack.config` refers to the `cfg` global variable.

```js
externals: "../webpack.config"
// is the same as:
externals: {
  "../webpack.config": "../webpack.config",
  //                    ^^^^^^^^^^^^^^^^^
  //                   invalid binding name
}
```

**tl;dr:** use object notation

**Comment by weijiangan** [comment]
@ooflorent Thanks for you explanation. 

I get you, but how does that help me exclude my webpack.config from the bundling process? Now it's referencing for this `cfg` global var which is undefined. My case is not the same as the example in the docs 
```js
externals: {
  jquery: 'jQuery'
}
```
where jQuery is loaded from a CDN and the global var is set

**Comment by eng-dibo** [comment]
up
